### PR TITLE
[GETandALGN]: Get channel and alignment details

### PR DESCRIPTION
**Pull request by luckynikki**

### CODE REVIEW

File: tests/src/texture/hipGetAlgntoffset2D.cpp [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+#include <iostream>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <hip/hip_runtime.h>
+#include "test_common.h"
+
+using namespace std;
+#define HIP_ENABLE_PRINTF
```
Comment (gargrahul) [inline tests/src/texture/hipGetAlgntoffset2D.cpp]: Why do you need this?

**Comment by luckynikki** [comment]
Implemented all the suggestions.

**Comment by luckynikki** [comment]
Hi Maneesh,
#514 is not a replacement for #500. The additional tests with the name: 1) hipGetAlgntoffset2D.cpp  and 2)  hipGetChanDesc.cpp are pushed through the new branch with the PR #514.
Yes, I am fine with the closure of #500.

**Comment by mangupta** [comment]
@gargrahul Could you review this again and see if it is fine or are there any more changes that are required in this PR.

File: tests/src/texture/hipGetChanDesc.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+#define C 8 //columns, width
+
+bool testResult = true;
+
+void runTest(void);
+
+int main(int argc, char** argv) {
+    runTest();
+
+    if (testResult) {
```
Comment (gargrahul) [inline tests/src/texture/hipGetChanDesc.cpp]: Better to make runTest() return testResult. By default it is set to true.

File: tests/src/texture/hipGetAlgntoffset2D.cpp [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+texture<int, hipTextureType2D,hipReadModeElementType> tex;
+
+bool testResult = true;
+
+void runTest(void);
+
+int main(int argc, char** argv) {
+    runTest();
+
+    if (testResult) {
```
Comment (gargrahul) [inline tests/src/texture/hipGetAlgntoffset2D.cpp]: Better to return testResult from runTest(). It is set to true by default.

File: tests/src/texture/hipGetAlgntoffset2D.cpp [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+size_t offset;
+
+for(i=0;i<R;i++)
+    for(j=0;j<C;j++)
+	{
+	 val[i][j]=(i+1)*(j+1);
+	}
+
+
+hipChannelFormatDesc chan_test,chan_desc=hipCreateChannelDesc(32,0,0,0,hipChannelFormatKindSigned);
```
Comment (gargrahul) [inline tests/src/texture/hipGetAlgntoffset2D.cpp]: chan_test is not used in this test. You can remove it.

**Comment by gargrahul** [comment]
@luckynikki Please confirm if these tests works well on HIP/NVCC too.

**Comment by luckynikki** [comment]
HI Rahul,
I have edited at appropriate lines as per your suggestions. These tests are excluded for nvcc for now.
Thanks.

**Comment by luckynikki** [comment]
Enable the two test cases(hipGetChanDesc.cpp and hipGetAlgntoffset2D.cpp) for nvcc also.

File: tests/src/texture/hipGetAlgntoffset2D.cpp [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+#include "test_common.h"
+
+using namespace std;
+#define R 8 //rows, height
+#define C 8 //columns, width
+
+
+texture<int, hipTextureType2D,hipReadModeElementType> tex;
+
+bool testResult = true;
```
Comment (gargrahul) [inline tests/src/texture/hipGetAlgntoffset2D.cpp]: This can be removed now.

File: tests/src/texture/hipGetChanDesc.cpp [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+ */
+
+#include <hip/hip_runtime.h>
+#include "test_common.h"
+
+using namespace std;
+#define R 8 //rows, height
+#define C 8 //columns, width
+
+bool testResult = true;
```
Comment (gargrahul) [inline tests/src/texture/hipGetChanDesc.cpp]: This can be removed now.

**Comment by gargrahul** [comment]
Thanks. Can you run these on NVCC path too since #563 is merged now?

I would like to test texture tests on both HIP/HCC and HIP/NVCC as much as possible.

**Pull request merged**